### DISCUSSION TITLE
Inserting a scrollbar

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [lc-soft/LCUI#247] [Q&A] [answered]
Hello, is the any special configuration when creating a scrollbar? From my understanding if my component layout is like:

- root
  - content
    - item1
    - item2
  - scrollbar

I should run `ScrollBar_BindTarget` for `scrollbar -> content`. And `Widget_Append content -> root`, then `scrollbar -> root` and it should work. However, I am not seeing the scrollbar in my component. Am I doing something wrong or is there a css property, which needs to be set for this to work?

**Edit**
* I tried to set `overflow: scroll` on content and root, but it still doesn't work.
* `ScrollBar_BindBox` doesn't solve it either

**Comment by WhoAteDaCake** [comment]
Hmm, it might be issue with the bindings I'm creating, even with trying to recreate **test_scrollbar.c** it's not working

**Reply by WhoAteDaCake** [reply]
Tried to use `LCUIBuilder_LoadFile` to avoid any changes, yet oddly enough, it displays the horizontal bar correctly, but not the vertical one 🤷  

<img width="1256" alt="20210612_175230_Remote Desktop" src="https://user-images.githubusercontent.com/16735813/121783573-4571e080-cba7-11eb-8ff0-3c53bc07cdbd.png">

**Comment by lc-soft** [comment]
>  However, I am not seeing the scrollbar in my component.

The scrollbar is displayed when the content size exceeds the container size.

If you want to always display the scroll bar, we need to modify the source code of the scroll bar, add a property to express always display the scroll bar, and then modify the if statement here.

https://github.com/lc-soft/LCUI/blob/85958e44051e60edcb2260f0fb8b1628219a61f0/src/gui/widget/scrollbar.c#L407-L416

> I tried to set overflow: scroll on content and root, but it still doesn't work.

![image](https://user-images.githubusercontent.com/1730073/121792530-9bd33380-cc28-11eb-8885-97793388cf7c.png)

**Comment by WhoAteDaCake** [comment, marked as answer]
Turns out the issue was because my parent containers did not have height property set correctly, after setting all of them to `height: 100%`, it worked as expected
